### ISSUE TITLE
F-Droid build questions

### ISSUE THREAD

**Issue by licaon-kter** [organicmaps/organicmaps#3800]
Updated the recipe: https://gitlab.com/fdroid/fdroiddata/-/commit/01bf93f506eafa2b87287529597a13809738f4ba

As the build log said: 
> ERROR: Could not build app app.organicmaps: Unexpected version/version code in output; APK: '2022.11.02-2' / '22110202',  Expected: '2022.11.02-2-FDroid' / '22110202'

Where did the suffix go?

**Comment by biodranik** [comment]
Where did you see it in the log? We did not change anything:
```
--- a/android/src/fdroid/play/version.yaml
+++ b/android/src/fdroid/play/version.yaml
@@ -1 +1 @@
-version: 2022.09.22-3-FDroid+22092203
+version: 2022.11.02-2-FDroid+22110202
```

**Comment by licaon-kter** [comment]
Local build: 
[app.organicmaps_22110202.log.gz](https://github.com/organicmaps/organicmaps/files/9918174/app.organicmaps_22110202.log.gz)

**Comment by linsui** [comment]
The suffix is there https://github.com/organicmaps/organicmaps/blob/2022.11.02-2-android/android/build.gradle#L258. But we removed it https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/app.organicmaps.yml#L1025.

**Comment by linsui** [comment]
By the way, I'd llike to split the abi for F-Droid. What version code scheme should we use?

**Comment by biodranik** [comment]
@linsui why did you remove the suffix? We use the suffix to support our users. Can you please revert it?

@linsui what exactly do you need for abi splitting and why? Won't it complicate everything?

**Comment by linsui** [comment]
We set the default version name to the one with suffix. I thought we did that because there was some bug. It fails this time because the code changed from `ver.second` to `ver.V2` in this version. Maybe we can remove the patch.

**Comment by licaon-kter** [comment]
@biodranik this change broke it https://github.com/organicmaps/organicmaps/commit/f8dec32d8f3b769e4ff5b4f05d160bdef646cd7e#diff-197b190e4a3512994d2cebed8aff5479ff88e136b8cc7a4b148ec9c3945bd65aL144

**Comment by linsui** [comment]
> what exactly do you need for abi splitting and why? Won't it complicate everything?

It can reduce the apk size. I thought we just need to add different gradle props for different abi. But we also need different version code for different apk.

**Comment by linsui** [comment]
Do we still need to build boost or we can use libboost-dev from debian? It's 1.74.

**Comment by biodranik** [comment]
> @biodranik this change broke it https://github.com/organicmaps/organicmaps/commit/f8dec32d8f3b769e4ff5b4f05d160bdef646cd7e#diff-197b190e4a3512994d2cebed8aff5479ff88e136b8cc7a4b148ec9c3945bd65aL144

Why do you think it is related? The behavior should be exactly the same. -FDroid is appended separately in another place:
```
    fdroid {
      dimension "default"
      versionName = android.defaultConfig.versionName + '-FDroid'
      ...
```

**Comment by biodranik** [comment]
@linsui we don't need boost libs. We use boost headers. They are properly generated by boost's b2 binary.

**Comment by biodranik** [comment]
> It can reduce the apk size. I thought we just need to add different gradle props for different abi. But we also need different version code for different apk.

The size can be reduced from 56Mb to 44Mb, do you think it is worth it? It's way easier for users to download the universal apk without worrying about the abi.

**Comment by licaon-kter** [comment]
See the recipe above, we sed remove all instances of `-FDroid` first and then in place replace those two strings (that changed) with version code and name

**Comment by linsui** [comment]
Yes, can we use the libboost-dev from debian? It's the headers. https://packages.debian.org/bullseye/amd64/libboost1.74-dev/filelist

**Comment by linsui** [comment]
> The size can be reduced from 56Mb to 44Mb, do you think it is worth it? It's way easier for users to download the universal apk without worrying about the abi.

Ah, this difference is small. Then let's keep it as is.

**Comment by biodranik** [comment]
> Yes, can we use the libboost-dev from debian? It's the headers. https://packages.debian.org/bullseye/amd64/libboost1.74-dev/filelist

We use the latest boost 1.80+, I prefer to avoid surprises in production. What are the benefits to use the outdated boost lib from the system?

**Comment by biodranik** [comment]
> See the recipe above, we sed remove all instances of -FDroid first and then in place replace those two strings (that changed) with version code and name

Sorry, I lost you. No patches should be necessary, we didn't change anything related to version naming in last several months.

**Comment by linsui** [comment]
> What are the benefits to use the outdated boost lib from the system?

It takes lots of time to fetch and built it. Not a big problem though.

**Comment by linsui** [comment]
> Sorry, I lost you. No patches should be necessary, we didn't change anything related to version naming in the last several months.

IIRC I patch it because in a version the version name from the txt file is different from the apk so the build failed. So I just force the version name the same as the one in our metadata instead of generating it. I thought we can remove the patch now.

**Comment by licaon-kter** [comment]
Updated the recipe to fix the patch: https://gitlab.com/fdroid/fdroiddata/-/commit/1af82f2f9bb34545e677186b01c83c2ac8d605d2

**Comment by biodranik** [comment]
Aha, now I understand why it failed. Thanks!

**Comment by licaon-kter** [comment]
Removed the patching: https://gitlab.com/fdroid/fdroiddata/-/commit/c2c4a198c12107700b666a2fba874b1c53b9fc0b

**Comment by biodranik** [comment]
What is this issue about? https://monitor.f-droid.org/builds/log/app.organicmaps/22110202#site-footer

**Comment by linsui** [comment]
It's a buildserver bug. It has been fixed. The app will be built later.
